test/test-lex.cpp

```diff
@@ -266,36 +377,50 @@ TEST(test_lex, lex_invalid_big_int_number) {
   }
 
   // Complain about both the decimal point and the leading 0 digit.
+  // q(👮🏽‍♀️)S
+  // o do we want this to be?  Should it complain it's a bad octal number
+  // (because of `.`) and bad big_int because of `.2n`?
+  // The problem with that is I lex the `.` in `lexer::parse_octal_number`
   {
     error_collector v;
     padded_string input(u8"01.2n");
     lexer l(&input, &v);
     EXPECT_EQ(l.peek().type, token_type::number);
     l.skip();
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
     EXPECT_EQ(l.peek().type, token_type::end_of_file);
 
     EXPECT_THAT(
         v.errors,
         UnorderedElementsAre(
-            VariantWith<error_big_int_literal_contains_decimal_point>(_),
-            VariantWith<error_big_int_literal_contains_leading_zero>(_)));
+            VariantWith<error_unexpected_characters_in_octal_number>(_)
+            /* VariantWith<error_big_int_literal_contains_decimal_point>(_), */
+            /* VariantWith<error_big_int_literal_contains_leading_zero>(_)*/
+            ));
   }
 
   // Complain about everything. What a disaster.
+  // q(👮🏽‍♀️)S
+  // o do we want this to be?  Should it complain it's a bad octal number
+  // (because of `.`) and bad big_int because of `.2n`?
+  // The problem with that is I lex the `.` in `lexer::parse_octal_number`
   {
     error_collector v;
     padded_string input(u8"01.2e+3n");
     lexer l(&input, &v);
     EXPECT_EQ(l.peek().type, token_type::number);
     l.skip();
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
     EXPECT_EQ(l.peek().type, token_type::end_of_file);
 
     EXPECT_THAT(
         v.errors,
         UnorderedElementsAre(
-            VariantWith<error_big_int_literal_contains_decimal_point>(_),
-            VariantWith<error_big_int_literal_contains_exponent>(_),
-            VariantWith<error_big_int_literal_contains_leading_zero>(_)));
+            VariantWith<error_unexpected_characters_in_octal_number>(_),
+            /* VariantWith<error_big_int_literal_contains_decimal_point>(_), */
```

Comment:
Not satisfied with this.  The test description says it should "Complain about everything", but I am getting rid of complaints